### PR TITLE
Fix Java path param dispatching

### PR DESCRIPTION
**Pull request by mmihic**
Java parameter types don't map directly onto created KTypes, so Java
applications attempting to use string conversion in e.g. PathParams or
FormParams would blow up unable to find an appropriate conversion
function. Fixed and added tests.

### CODE REVIEW

File: exemplar/src/main/java/com/squareup/exemplar/package-info.java [hunk 1]
```diff
@@ -0,0 +1,4 @@
+@ParametersAreNonnullByDefault
+package com.squareup.exemplar;
+
+import javax.annotation.ParametersAreNonnullByDefault;
```
Comment (swankjesse) [inline exemplar/src/main/java/com/squareup/exemplar/package-info.java]: nice

File: misk/src/test/java/misk/web/extractors/JavaPathParamDispatchTest.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  }
+
+  public static final class GetObjectDetails implements WebAction {
+    @Get(pathPattern = "/objects/{objectType}/{name}/{version}")
+    @ResponseContentType(MediaTypes.TEXT_PLAIN_UTF8)
+    public String getObjectDetails(
+        @PathParam("objectType") ObjectType objectType,
+        @PathParam("name") String name,
+        @PathParam("version") long version
+    ) {
```
Comment (swankjesse) [inline misk/src/test/java/misk/web/extractors/JavaPathParamDispatchTest.java]: the Kotlin is working on you

File: misk/src/test/java/misk/web/extractors/JavaPathParamDispatchTest.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+import okhttp3.Request;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@MiskTest(startService = true)
+public class JavaPathParamDispatchTest {
+  @MiskTestModule final Module module = new TestModule();
+
+  public enum ObjectType {
```
Comment (swankjesse) [inline misk/src/test/java/misk/web/extractors/JavaPathParamDispatchTest.java]: ResourceType ?

ObjectType sounds like java.lang.Object which was threw me off for a sec

File: misk/src/main/kotlin/misk/web/extractors/StringConverter.kt [hunk 4]
```diff
@@ -2,35 +2,29 @@ package misk.web.extractors
...
+internal fun converterFor(type: KType): StringConverter? {
+
+  return when {
+    type.isSubtypeOf(stringType) -> { it -> it }
+    type.isSubtypeOf(intType) -> numberConversionWrapper { it.toInt() }
+    type.isSubtypeOf(longType) -> numberConversionWrapper { it.toLong() }
+    type.isSubtypeOf(doubleType) -> numberConversionWrapper { it.toDouble() }
+    type.isSubtypeOf(booleanType) -> numberConversionWrapper { it.toBoolean() }
+    else -> createFromValueOf(type)
+  }
```
Comment (eeSeeGee) [inline misk/src/main/kotlin/misk/web/extractors/StringConverter.kt]: nice

**Pull request merged**